src/App.js

```diff
@@ -1,122 +1,99 @@
+      }
+    };
+    document.addEventListener("keydown", handleKeyDown);
+    return () => {
+      document.removeEventListener("keydown", handleKeyDown);
     };
-    return (
-        <div className="App">
-            <div className="title-section">
-                <h1>Poke Stats</h1>
-                <input
-                    className="search-input"
-                    type="text"
-                    onChange={(event) => {
-                        setPokemonName(event.target.value);
-                    }}
-                ></input>
-                <button onClick={searchPokemon} className="search-button">
-                    Search Pokemon
-                </button>
-            </div>
-            <div className="display-section">
-                {!isChosen ? (
-                    <h1>Please choose a pokemon</h1>
-                ) : (
-                    <div className="stats-window">
-                        <h1 className="stats-title">{pokemonInfo.name}</h1>
-                        <div className="text-and-image-wrapper">
-                            <div className="text-stats">
-                                <p>
-                                    <span className="bold">Species:</span>{" "}
-                                    {pokemonInfo.species}
-                                </p>
-                                <p>
-                                    <span className="bold">Type:</span>{" "}
-                                    {pokemonInfo.type}
-                                </p>
-                                <p>
-                                    <span className="bold">HP:</span>{" "}
-                                    {pokemonInfo.hp}
-                                </p>
-                                <p>
-                                    <span className="bold">Attack:</span>{" "}
-                                    {pokemonInfo.attack}
-                                </p>
-                                <p>
-                                    <span className="bold">Defence:</span>{" "}
-                                    {pokemonInfo.defense}
-                                </p>
-                            </div>
-                            <div className="image-window">
-                                <img
-                                    className="pokemon-image"
-                                    src={pokemonInfo.imageUrl}
-                                    alt="front view of chosen pokemon"
-                                ></img>
-                            </div>
-                        </div>
-                    </div>
-                )}
-            </div>
-        </div>
-    );
+  }, []);
+
+  const handleChosen = () => {
+    if (pokemonName) {
+      setIsChosen(true)
+    }
+  }
+
+  useEffect(() => {
+    if (location.pathname !== "/") {
+      setIsChosen(true)
+    } else {
+      setIsChosen(false);
+    }
+  }, [location.pathname]);
```

Comment:
If you use the `index` route you don't need _any_ of this code `cause you can rely on the URL and react router to tell you if a pokemon has been chosen or not based on the URL

i.e React router will look at the URL and decide which component to render:

| path      | state |
| ----------- | ----------- |
| `/`      | not chosen (render PokemonIndexPage)       |
| `/:pokemonName`   | chosen (render PokemonDetailsPage        |